### PR TITLE
Referral Link in TopNav on Desktop

### PR DESCRIPTION
**Pull request by youngkidwarrior**

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
* **#318** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/318?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#317** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/317?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @youngkidwarrior and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/app/components/ReferralLink.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import { IconCopy } from './icons'
+import * as Clipboard from 'expo-clipboard'
+import * as Sharing from 'expo-sharing'
+
+export function ReferralLink() {
+  const { profile } = useUser()
+  const referralCode = profile?.referral_code
+  const primaryTag = useConfirmedTags()?.at(0) ?? profile?.referral_code
+  const referralHref = `https://send.app/${referralCode}`
+  const visualReferralUrl = `https://send.app/${primaryTag}`
```
Comment (0xBigBoss) [inline packages/app/components/ReferralLink.tsx:15]: this is misleading to build the link like this. I know the design has it that way (probably). 

Let's just have it link `https://send.app/` or have the text more clear `my referral link`

File: packages/app/components/ReferralLink.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import { CheckCheck, CheckCircle } from '@tamagui/lucide-icons'
+import { useEffect, useState } from 'react'
+import { IconCopy } from './icons'
+import * as Clipboard from 'expo-clipboard'
+import * as Sharing from 'expo-sharing'
+
+export function ReferralLink() {
+  const { profile } = useUser()
+  const referralCode = profile?.referral_code
+  const primaryTag = useConfirmedTags()?.at(0) ?? profile?.referral_code
```
Comment (0xBigBoss) [inline packages/app/components/ReferralLink.tsx:13]: also this is an object not a sendtag name.

Screenshot 2024-04-15 at 21 49 55

**Review by 0xBigBoss — CHANGES_REQUESTED** [review]

File: packages/app/components/ReferralLink.tsx (hunk 1 above)
Comment (youngkidwarrior) [reply packages/app/components/ReferralLink.tsx]: Sure thing, I'll just show the referral code for now

File: packages/app/components/ReferralLink.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import { CheckCheck } from '@tamagui/lucide-icons'
+import { useEffect, useState } from 'react'
+import { IconCopy } from './icons'
+import * as Clipboard from 'expo-clipboard'
+import * as Sharing from 'expo-sharing'
+
+export function ReferralLink() {
+  const { profile } = useUser()
+  const referralCode = profile?.referral_code
+  const referralHref = `https://send.app/${referralCode}`
```
Comment (0xBigBoss) [inline packages/app/components/ReferralLink.tsx:12]: `https://send.app/${referralCode}` would make an href like this `https://send.app/myreferralcode`? It's not a valid referral code.

File: packages/app/components/TopNav.tsx [hunk 4]
```diff
@@ -131,12 +147,17 @@ export function TopNav({
               $theme-light={{ col: '$gray10Light' }}
               $theme-dark={{ col: '$gray8Light' }}
               lineHeight={32}
-              $gtLg={{ ml: isSubRoute ? '$4' : '$0' }}
+              // $gtLg={{ display: 'none' }}
```
Comment (0xBigBoss) [inline packages/app/components/TopNav.tsx:150]: why was this needed?

File: packages/app/components/ReferralLink.tsx (hunk 3 above)
Comment (youngkidwarrior) [reply packages/app/components/ReferralLink.tsx:12]: O shit. It's a query param. Designs threw me off

File: packages/app/components/TopNav.tsx (hunk 4 above)
Comment (youngkidwarrior) [reply packages/app/components/TopNav.tsx:150]: unintenional

File: packages/app/components/ReferralLink.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            theme: 'error',
+          },
+        })
+      )
+  }
+
+  useEffect(() => {
+    if (hasCopied) {
+      setTimeout(() => {
+        setHasCopied(false)
```
Comment (0xBigBoss) [inline packages/app/components/ReferralLink.tsx:45]: this isn't working for me for some reason.

image

https://github.com/0xsend/sendapp/assets/95193764/ce6d825d-917e-47c2-aace-93af6d657da0

**Review by 0xBigBoss — APPROVED** [review]

File: packages/app/components/ReferralLink.tsx (hunk 5 above)
Comment (youngkidwarrior) [reply packages/app/components/ReferralLink.tsx:45]: Haha weird. I'm going to merge for now, since not a big deal. We can fix bug in future PR

**Comment by youngkidwarrior** [comment]
### Merge activity

* **Apr 18, 6:49 PM EDT**: @youngkidwarrior started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/0xsend/sendapp/317).

**Pull request merged**